assignment.rb

```diff
@@ -0,0 +1,295 @@
+    when 3
+      # The user would like to exit the program entirely.
+      @exit_prog = true
+      puts "Now exiting the ATM program. Goobye."
+      puts ""
+
+    # This else should be theoretically impossible. 
+    else
+      display_error()
+    end
+  end
+
+  # We need to access @exit_prog's value outside of the class. Let's make a getter
+  def exit_prog
+    @exit_prog
+  end
+
+  # This method displays input prompt and returns the value input as a string. 
+  # Loops until input is given.
+  def get_input(prompt)
+    var = ""
+    while var == ""
+      puts "#{prompt}"
+      print ">> "
+      var = gets.chomp
+      puts ""
+    end
+
+    var
+  end
+
+  # Gathers the user's login and pin information
+  def get_login_info
+    name = get_input("Please enter your name")
+    pin = get_input("Please enter your pin")
+    login = { name: name, pin: pin}
+  
+   login
+  end
+
+  # This method displays a menu and verifies that the 
+  # input collected and returned is a valid option. 
+  # Menu must be an array with options starting at 1.
+  def get_menu_selection(menu)
+    men_sel = 0
+    while !(1..menu.count).include?(men_sel)
+      men_sel = get_input(menu.join("\n")).to_i
+      if !(1..menu.count).include?(men_sel)
+        puts "Invalid entry. Try again."
+        puts ""
+      end
+    end
+
+    men_sel
+  end
+
+  # This method takes login info and searches the "database" for a valid
+  # user id and pin to match. If no match is found, valid user is left false.
+  def get_user(login)
+    match_found = false
+    @account_holders.each do |user|
+      if user.name == login[:name] && user.pin == login[:pin]
+        @show_acc_menu = true
+        match_found = true
+        puts "Your account has been verified."
+        puts ""
+        return user
+      end
+    end
+    if match_found == false
+      puts "Something went wrong. Please start over."
+      puts ""
+    end
+  end
+  
+  def display_error
+    puts "Something has gone horribly wrong. This machine"
+    puts "is now set to self detonate. You should run."
+    puts ""
+  end
+
+  # This method is responsible for operating the Account Holder's menu
+  def show_acc_menu
+    # Now display the account menu and get the user's input
+    acc_menu_sel = get_menu_selection(ACC_MENU)
+
+    # Depending on the user's input, perform the correct function
+    case acc_menu_sel
+    # Account Menu option 1 indicates the user wants to check their balance.
+    when 1
+      puts "Your current account balance is: $#{@user.balance}"
+      puts ""
+    # Account Menu option 2 indicates the user would like to withdraw funds
+    when 2
+      # Make a withdrawal
+      withdraw_funds()
+    # Account Menu option 3 indicates the user would like to log out
+    when 3
+      @show_acc_menu = false
+      puts "Returning to the Main Menu."
+      puts ""
+    # If menu options 1-3 have failed, the world might end. Display an error
+    else
+      display_error()
```

Comment:
I prefer calling the empty parens to help make it clear this is a method, not a variable. Unless it against the Ruby style, I'll keep these in place.